### PR TITLE
DOPE-261: change like to support usage with TypeExpression

### PR DESCRIPTION
**Pull request by jansigi**

### CODE REVIEW

File: crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt [hunk 1]
```diff
@@ -16,20 +18,86 @@ class LikeTest : ManagerDependentTest {
...
+        val right = someCMStringField()
+        val expected = LikeExpression(left.toDopeType(), right.toDopeType())
+
+        val actual = left.isLike(right)
+
+        assertEquals(expected.toDopeQuery(manager), actual.toDopeQuery(manager))
+    }
+
+    @Test
+    fun `should support extensions for is like with cmString`() {
```
Comment (martinagallati-ergon) [inline crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt:43]: test name should end with "...like with cmString string"

File: crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt [hunk 2]
```diff
@@ -16,20 +18,86 @@ class LikeTest : ManagerDependentTest {
...
+        val right = someString()
+        val expected = LikeExpression(left.toDopeType(), right.toDopeType())
+
+        val actual = left.isLike(right)
+
+        assertEquals(expected.toDopeQuery(manager), actual.toDopeQuery(manager))
+    }
+
+    @Test
+    fun `should support extensions for is like with cm`() {
```
Comment (martinagallati-ergon) [inline crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt:54]: would change it to "...like with cmString cmString"

File: crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt [hunk 3]
```diff
@@ -16,20 +18,86 @@ class LikeTest : ManagerDependentTest {
...
 
-        val actual = someCMStringField.isLike(right)
+        val actual = left.isNotLike(right)
 
         assertEquals(expected.toDopeQuery(manager), actual.toDopeQuery(manager))
     }
 
     @Test
-    fun `should support extensions for is not like with CMJsonField`() {
+    fun `should support extensions for is not like with cm`() {
```
Comment (martinagallati-ergon) [inline crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt:98]: see above

**Review by martinagallati-ergon — APPROVED** [review]

File: crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt (hunk 1 above)
Comment (jansigi) [reply crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt]: done

File: crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt (hunk 2 above)
Comment (jansigi) [reply crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt]: done

File: crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt (hunk 3 above)
Comment (jansigi) [reply crystal-map-connector/src/test/kotlin/ch/ergon/dope/extensions/type/relational/LikeTest.kt]: done

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/LikeExpression.kt [hunk 4]
```diff
@@ -2,31 +2,37 @@ package ch.ergon.dope.resolvable.expression.unaliased.type.relational
...
     override fun toDopeQuery(manager: DopeQueryManager) = toInfixDopeQuery(manager = manager)
 }
 
-fun Field<out ValidType>.isLike(right: TypeExpression<StringType>) = LikeExpression(this, right)
+fun TypeExpression<StringType>.isLike(right: TypeExpression<StringType>) = LikeExpression(this, right)
 
-fun Field<out ValidType>.isLike(right: String) = isLike(right.toDopeType())
+fun TypeExpression<StringType>.isLike(right: String) = isLike(right.toDopeType())
+
+fun String.isLike(right: TypeExpression<StringType>) = toDopeType().isLike(right)
```
Comment (pgruntz) [inline core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/LikeExpression.kt:21]: please add some buildTest, where you pass more complex type-expression (e.g. concat, like it is described in the ticket)

**Review by pgruntz — APPROVED** [review]

**Pull request merged**